nominal/core.py

```diff
@@ -311,6 +322,174 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
         )
 
 
+# TODO(ritwikdixit): add support for more Checklist metadata and versioning
+@dataclass(frozen=True)
+class Checklist:
+    rid: str
+    title: str
+    description: str
+    properties: Mapping[str, str]
+    labels: Sequence[str]
+    variables: Sequence[ChecklistVariable]
+    checks: Sequence[Check]
+    _client: NominalClient = field(repr=False)
+
+    # we do this afterward to batch the request
+    def _resolve_expressions(self) -> None:
+        variable_expressions = self._client._compute_representation_client.batch_compute_to_expression(
+            self._client._auth_header,
+            {var.name: var.compute_graph for var in self.variables}
+        )
+        check_expressions = self._client._compute_representation_client.batch_compute_to_expression(
+            self._client._auth_header,
+            {check.rid : check.compute_graph for check in self.checks}
```

Comment:
we send out 2 network requests here, could be done in one, but I think it's fine